### PR TITLE
Github Code: Initial sync workflow integration

### LINKED ISSUE #3102: [Caloris2] [Github Code] integration with initial sync workflows

### PR DESCRIPTION
**Pull request by spolu**
Fixes: #3102

```
- Tested initial sync of repos (sync-repo admin cli)
- Tested re-running (everything is correctly skipped)
- Tested re-running with an update to the repo (only the updated file gets updated)

- Tested global `code-sync --enable true` (admin cli) 
- Tested global `code-sync --enable false` (admin cli) (everything gets clean-ed as expected)
```

When this merge we can enable/disable code sync on our own github (no incremental sync) or other users with
```
./admin/cli.sh github code-sync --wId 9279eb4dc1 --dataSourceName managed-github --enable true
./admin/cli.sh github code-sync --wId 9279eb4dc1 --dataSourceName managed-github --enable false
```

Subsequent PRs:
- Concurrency upserting and deleting
- Incremental sync

Deploy:
- deploy to connectors-edge
- init_db (since I had forgotten to add the new models)
- deploy to connectors

### CODE REVIEW

File: connectors/src/connectors/github/index.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export async function createGithubConnector(
 
       return connector;
     });
-    await launchGithubFullSyncWorkflow(connector.id.toString());
+    await launchGithubFullSyncWorkflow(connector.id.toString(), false);
```
Comment (fontanierh) [inline connectors/src/connectors/github/index.ts:62]: can we make it an object params ? the dangling booleans are hard to read IMO

File: connectors/src/connectors/github/temporal/workflows.ts [hunk 2]
```diff
@@ -40,19 +40,26 @@ const { githubUpsertIssueActivity, githubUpsertDiscussionActivity } =
...
+  connectorId: string,
+  // Used to re-trigger a code-only full-sync after code syncing is enabled/disabled.
+  syncCodeOnly: boolean
 ) {
   await githubSaveStartSyncActivity(dataSourceConfig);
   const loggerArgs = {
     dataSourceName: dataSourceConfig.dataSourceName,
     workspaceId: dataSourceConfig.workspaceId,
     githubInstallationId,
+    syncCodeOnly: syncCodeOnly ? "true" : "false",
```
Comment (fontanierh) [inline connectors/src/connectors/github/temporal/workflows.ts:62]: nit: `syncCodeOnly.toString()`

File: connectors/src/connectors/github/temporal/workflows.ts [hunk 3]
```diff
@@ -144,87 +155,128 @@ export async function githubRepoSyncWorkflow(
...
-            loggerArgs,
-            true // isBatchSync
+      if (!resultsPage.length) {
+        break;
+      }
+      pageNumber += 1;
+      for (const issueNumber of resultsPage) {
+        promises.push(
+          queue.add(() =>
+            githubUpsertIssueActivity(
```
Comment (fontanierh) [inline connectors/src/connectors/github/temporal/workflows.ts:193]: I guess this one should also have been an object params

**Review by fontanierh — APPROVED** [review]
LGTM

File: connectors/src/connectors/github/index.ts (hunk 1 above)
Comment (spolu) [reply connectors/src/connectors/github/index.ts:62]: def

File: connectors/src/connectors/github/temporal/workflows.ts (hunk 3 above)
Comment (spolu) [reply connectors/src/connectors/github/temporal/workflows.ts:193]: will change :+1:

Comment (fontanierh) [reply connectors/src/connectors/github/temporal/workflows.ts:193]: Yes 😅  I wrote this one

Comment (fontanierh) [reply connectors/src/connectors/github/temporal/workflows.ts:193]: I'll make a pass to cleanup all others 👍

**Review by fontanierh — APPROVED** [review]

**Pull request merged**